src/ArrayFromIterator.php

```diff
@@ -43,11 +43,7 @@
     final private function all() : PromiseSequence
     {
         return new PromiseSequence(
-            $promise = new Promise(
-                function () use (&$promise) {
-                    $promise->resolve(new ArraySequence($this->toArray()));
-                }
-            )
+            new FulfilledPromise(new ArraySequence($this->toArray()))
```

Comment:
you may even be able to pass a promise of an array, the constructor of `PromiseSequence` seems to accept it